File: Habitica/src/main/java/com/habitrpg/android/habitica/ui/views/insufficientCurrency/InsufficientCurrencyDialog.java
package com.habitrpg.android.habitica.ui.views.insufficientCurrency;

import android.content.Context;
import android.support.v7.app.AlertDialog;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import com.habitrpg.android.habitica.R;

import butterknife.BindView;
import butterknife.ButterKnife;

/**
 * Created by phillip on 27.09.17.
 */

abstract public class InsufficientCurrencyDialog extends AlertDialog {

    @BindView(R.id.imageView)
    ImageView imageView;
    @BindView(R.id.textView)
    TextView textView;

    public InsufficientCurrencyDialog(Context context) {
        super(context);

        LayoutInflater inflater = LayoutInflater.from(context);
        View view = inflater.inflate(R.layout.dialog_insufficient_currency, null);
        ButterKnife.bind(this, view);
        setView(view);

        this.setButton(AlertDialog.BUTTON_NEUTRAL, context.getString(R.string.close), (dialogInterface, i) -> {
            this.dismiss();
        });
    }

}

File: Habitica/src/main/java/com/habitrpg/android/habitica/ui/views/insufficientCurrency/InsufficientCurrencyDialog.kt
package com.habitrpg.android.habitica.ui.views.insufficientCurrency

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.widget.ImageView
import android.widget.TextView
import com.habitrpg.android.habitica.R
import com.habitrpg.android.habitica.ui.views.dialogs.HabiticaAlertDialog

/**
 * Created by phillip on 27.09.17.
 */

abstract class InsufficientCurrencyDialog(context: Context) : HabiticaAlertDialog(context) {
    protected lateinit var imageView: ImageView
    protected lateinit var textView: TextView

    open fun getLayoutID(): Int {
        return R.layout.dialog_insufficient_currency
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val inflater = LayoutInflater.from(context)
        val view = inflater.inflate(getLayoutID(), null)
        setAdditionalContentView(view)

        imageView = view.findViewById(R.id.imageView)
        textView = view.findViewById(R.id.textView)
    }
}
